### DISCUSSION TITLE
selectWithinFam() fails if a parent is set to unknown

### DISCUSSION THREAD

**Discussion by JanezJ2** [gaynorr/AlphaSimR#71] [Ideas]
**Describe the point**
We noticed that `selectWithinFam()` fails if a parent is set to unknown.

**To Reproduce**
```
#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=10)

#Set simulation parameters
SP = SimParam$new(founderPop)
SP$addTraitA(10)
SP$setVarE(h2=0.5)

#Create population
pop = newPop(founderPop, simParam=SP)

#Create 3 biparental families with 10 progeny
pop2 = randCross(pop, nCrosses=3, nProgeny=10, simParam=SP)

#Select best individual per family
selectWithinFam(pop2, 1, simParam=SP)
# --> returns 3 individuals

pop2@father[] = 0
pop2@mother[] = 0
selectWithinFam(pop2, 1, simParam=SP)
# --> returns 1 individual
```

**Expected behavior**
Obviously the above example is extreme and we asked for the trouble ourselves since we modified the father/mother information. However, if we move to real world scenarios then we can have missing pedigree data and then an example like above can become real - I am overstretching the argument here (we should have unknown parents only in the founderPop and not in the actual populations on which we run selection.

Happy for this discussion to be closed - the aim here is to share this experience.

**Comment by gaynorr** [comment]
Thanks for the call out. I hadn't thought about anyone trying to run the function on the founder population. I'll take a look at as part of the next update and see if I can come up with something sensible.
